test/cpp/tensorexpr/test_registerizer.cpp

```diff
@@ -0,0 +1,543 @@
+#include "test/cpp/tensorexpr/test_base.h"
+
+#include "test/cpp/tensorexpr/test_utils.h"
+#include "torch/csrc/jit/tensorexpr/ir_simplifier.h"
+#include "torch/csrc/jit/tensorexpr/registerizer.h"
+
+#include <iostream>
+
+namespace torch {
+namespace jit {
+using namespace torch::jit::tensorexpr;
+
+// Can replace a simple scalar access with a local variable.
+void testRegisterizerSimple() {
+  KernelScope kernel_scope;
+  Buffer a(BufHandle("A", {1}, kInt));
+  VarHandle x("x", kInt);
+  Stmt* stmt =
+      Block::make({Store::make(a, {0}, 0, 1),
+                   For::make(
+                       x,
+                       0,
+                       10,
+                       Block::make({Store::make(
+                           a, {0}, Add::make(Load::make(a, {0}, 1), x), 1)}))});
+
+  /*
+   * A[0] = 0;
+   * for (int x = 0; x < 10; x++) {
```

Comment:
Even in this case, please make sure you cannot replace the registers, if "x" is marked with threadIdx/blockIdx, or in the future, "paralellize".

Reply:
I am assuming that this pass occurs after block/thread axes are flattened down, but I'll add a check to make this explicit.